src/main/java/repository/Repository.java

```diff
@@ -0,0 +1,18 @@
+package repository;
+
+import java.nio.file.Path;
+import java.util.List;
+
+public interface Repository {
```

Comment:
I think this interface is fine (and what I was expecting you to do when I asked you to figure out how to unit test your file related functionality without actually creating files/directories), but ultimately if you were to write classes for other kinds of repositories, like a database, this interface wouldn't quite make sense. (you'd probably want methods for creating, reading, updating, removing, and querying). However, it would not make sense for you to change this interface at this point because you have not been asked to integrate a database and it would be wrong/too early to try to predict what this abstraction should look like. So no need to make any changes - I just wanted to point this out because I think `Repository` is very generic, and these functions are still quite specifically for a file system.